app/src/main/java/com/codesoom/assignment/service/ProductService.java

```diff
@@ -0,0 +1,59 @@
+package com.codesoom.assignment.service;
+
+import com.codesoom.assignment.Exception.ErrorCode;
+import com.codesoom.assignment.Exception.ProductException;
+import com.codesoom.assignment.domain.Product;
+import com.codesoom.assignment.domain.ProductJpaRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * responsible for the core business logics of Product Entity
+ */
+@Service
+public class ProductService {
+    private final ProductJpaRepository productJpaRepository;
+
+    public ProductService(ProductJpaRepository productJpaRepository) {
+        this.productJpaRepository = productJpaRepository;
+    }
+
+    /**
+     * returns a list of all products.
+     * @return a list of products
+     */
+    public List<Product> findAll() {
+        return productJpaRepository.findAll();
+    }
+
+    /**
+     * searches a product given the id
+     * @param id the Long to find a product
+     * @return Product entity; otherwise, throw an exception
+     */
```

Comment:
```suggestion
    /**
     * searches a product given the id
     * @param id the Long to find a product
     * @return Product entity
     * @throws 상품을 찾지 못한 경우
     */
```

JavaDoc을 학습하기 가장 좋은 예제는 Java의 빌트인 라이브러리입니다. 다음은 ArrayList 의 내용 중 일부에요.

![image](https://user-images.githubusercontent.com/1855714/194078316-d02d6af7-0be9-4dfd-86c1-ec8349b517d4.png)